### DISCUSSION TITLE
JIT compiler for JSImpl AST

### DISCUSSION THREAD

**Discussion by dtrendafilov** [SofiaCPP/IPL#107] [Projects]
Create a JIT compiler that from the AST (or bytecode of Spasm) of JSImpl produces native code for x86_64.

The compiler should work on function level and create a native version of the function given the types of the arguments - 32-bit integers and `float`s, 64-bit integers and `double`s.

- assume that the function will not access any variables from the outer scope.

Possible Optimizations:
- inlining
- loop unrolling
- branch elimination (provided with argument values)

**Comment by bobivk** [comment]
Ще се опитам да направя този проект за Windows.

**Comment by Xearty** [comment]
Ще правя този проект
